### PR TITLE
New device: Add sc7180-trogdor-lazor-limozeen with basic tests

### PR DESCRIPTION
**Pull request by kholk**
This patchset adds Trogdor to the test-configs and enables baseline and cros-ec tests to be run on it.

The baseline-nfs and other tests requiring nfs support haven't been added because of an issue with USB: it's probing slow (it's a module), the usb ethernet adapter comes up too late and this produces the unability to assign an IP address and mount the NFS rootfs.

Please note: in order to be able to run these tests, it is required to have this kernel patch merged in https://lore.kernel.org/lkml/20211011154003.904355-1-angelogioacchino.delregno@collabora.com
...or the device will hang.

### CODE REVIEW

File: config/core/test-configs.yaml [hunk 1]
```diff
@@ -1398,6 +1398,11 @@ device_types:
       - blocklist: *allmodconfig_filter
       - blocklist: {kernel: ['v3.', 'v4.4', 'v4.9']}
 
+  sc7180-trogdor-lazor-limozeen:
```
Comment (gctucker) [inline config/core/test-configs.yaml:1401]: Please send upstream in LAVA: https://git.lavasoftware.org/lava/lava/-/tree/master/etc/dispatcher-config/device-types

**Review by gctucker — COMMENTED** [review]
The devices are all in "looping" mode at the moment in the lab so no staging job has been run on them yet.  Let's take a look again tomorrow.
https://lava.collabora.co.uk/scheduler/device_type/sc7180-trogdor-lazor-limozeen?dt_length=25&dt_search=staging#dt_

File: config/core/test-configs.yaml (hunk 1 above)
Comment (kholk) [reply config/core/test-configs.yaml:1401]: Requested fork access. Will send as soon as they'll give it.

Comment (kholk) [reply config/core/test-configs.yaml:1401]: @gctucker https://git.lavasoftware.org/lava/lava/-/merge_requests/1612

Comment (kholk) [reply config/core/test-configs.yaml:1401]: @gctucker I had to create a new MR, which was almost instantly merged.

https://git.lavasoftware.org/lava/lava/-/merge_requests/1619

File: config/core/test-configs.yaml [hunk 2]
```diff
@@ -2410,6 +2415,11 @@ test_configs:
     test_plans:
       - baseline
 
+  - device_type: sc7180-trogdor-lazor-limozeen
+    test_plans:
+      - baseline
```
[the remark this answers is not in the document]
Comment (gctucker) [reply config/core/test-configs.yaml:2420]: @kholk It seems to be failing again: https://staging.kernelci.org/test/case/id/61930107a8a33eb09478a731/

Any news about your defconfig patches?

[the remark this answers is not in the document]
Comment (kholk) [reply config/core/test-configs.yaml:2420]: unfortunately no, I haven't received any reply on the defconfig patches... perhaps I should ping/resend...

[the remark this answers is not in the document]
Comment (gctucker) [reply config/core/test-configs.yaml:2420]: Yeah also I believe they need to be rebased, I had to rebase them so they would apply to the staging-mainline branch.  Some other changes went in during the merge window in the same area of the defconfig file.

**Comment by gctucker** [comment]
The config changes are being added to the `arm64-chromebook` fragment with #971

**Comment by gctucker** [comment]
@kholk The dtb names are changing in `linux-next` and there are several variants.  Which one should be used now for this particular platform?

**Comment by kholk** [comment]
> @kholk The dtb names are changing in `linux-next` and there are several variants. Which one should be used now for this particular platform?

for our Trogdor machines: sc7180-trogdor-lazor-limozeen-nots-r5.dtb

File: config/core/test-configs.yaml [hunk 3]
```diff
@@ -1398,6 +1398,13 @@ device_types:
       - blocklist: *allmodconfig_filter
       - blocklist: {kernel: ['v3.', 'v4.4', 'v4.9']}
 
+  sc7180-trogdor-lazor-limozeen:
+    mach: qcom
+    class: arm64-dtb
+    boot_method: depthcharge
+    dtb: 'sc7180-trogdor-lazor-limozeen-nots-r5.dtb'
```
Comment (gctucker) [inline config/core/test-configs.yaml:1405]:
```suggestion
    dtb: 'qcom/sc7180-trogdor-lazor-limozeen-nots-r5.dtb'
```

**Comment by gctucker** [comment]
OK that nearly worked, it got to a login prompt but the kernel was already on fire by then:
https://staging.kernelci.org/test/plan/id/61e868f371cbfcd785217814/
https://storage.staging.kernelci.org/kernelci/staging-mainline/staging-mainline-20220119.2/arm64/defconfig+arm64-chromebook+crypto/gcc-10/lab-collabora/baseline-sc7180-trogdor-lazor-limozeen.html

That was with mainline, let's see if linux-next behaves any better.

**Comment by gctucker** [comment]
Same problem on linux-next apparently:
https://storage.staging.kernelci.org/kernelci/staging-next/staging-next-20220119.6/arm64/defconfig+arm64-chromebook/gcc-10/lab-collabora/baseline-sc7180-trogdor-lazor-limozeen.html

**Comment by gctucker** [comment]
@kholk The errors mentioned above seem to be purely kernel issues rather than KernelCI configuration.  So I think we should have this PR merged and then the kernel issues can be addressed separately to get it to boot properly.

**Comment by gctucker** [comment]
Squashed the FIXUP commits and pushed the changes by hand to the `main` branch.

**Pull request closed without merging**